File: solutions/fsharp/bob/1/Bob.fs
module Bob

let response (input: string) : string =
    let isQuestion (input: string) = input.EndsWith '?'

    let isYelling (input: string) =
        Seq.exists System.Char.IsLetter input && input.ToUpper() = input

    match input.Trim() with
    | s when isQuestion s ->
        if isYelling s then
            "Calm down, I know what I'm doing!"
        else
            "Sure."
    | s when isYelling s -> "Whoa, chill out!"
    | "" -> "Fine. Be that way!"
    | _ -> "Whatever."

File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
  if (message.match(/^.*[A-Z].*$/)&& !message.match(/^.*[a-z].*$/) && message.match(/^.*\?\s*$/))
    return "Calm down, I know what I'm doing!"
  if (message.match(/^.*\?\s*$/))
    return "Sure."
  if (message.match(/^.*[A-Z].*$/)&& !message.match(/^.*[a-z].*$/))
    return "Whoa, chill out!"
  if (message.match(/^\s*$/))
    return "Fine. Be that way!"
  return "Whatever."
}
